### ISSUE TITLE
Another Promise Warning

### ISSUE THREAD

**Issue by funnylookinhat** [tgriesser/knex#1531]
I believe this should fix it - but I'm having trouble getting the tests running locally so I wanted to verify with you all before I created a PR.

https://github.com/tgriesser/knex/compare/master...funnylookinhat:promise-rejected-undefined?expand=1

Here's the stack I'm getting:

```
Warning: a promise was rejected with a non-error: [object Undefined]
  at /home/nexus/node_modules/knex/lib/dialects/mysql/transaction.js:49:27
  at bound (domain.js:287:14)
  at PassThroughHandlerContext.runBound (domain.js:300:12)
From previous event:
  at /home/nexus/node_modules/knex/lib/transaction.js:80:12
From previous event:
  at Transaction_MySQL.Transaction (/home/nexus/node_modules/knex/lib/transaction.js:51:40)
  at new Transaction_MySQL (/home/nexus/node_modules/knex/lib/dialects/mysql/transaction.js:30:28)
  at Client.transaction (/home/nexus/node_modules/knex/lib/client.js:171:12)
  at Function.transaction (/home/nexus/node_modules/knex/lib/util/make-knex.js:69:21)
  at /home/nexus/app/actions/index.coffee:209:12
From previous event:
  at UserUpdateValidate.BaseAction.executeExternal (/home/nexus/app/actions/index.coffee:208:9)
  at /home/nexus/app/actions/index.coffee:157:10
From previous event:
  at /home/nexus/app/actions/index.coffee:151:8
From previous event:
  at UserUpdateValidate.BaseAction.execute (/home/nexus/app/actions/index.coffee:137:9)
  at /home/nexus/app/actions/index.coffee:78:12
From previous event:
  at module.exports.runAction (/home/nexus/app/actions/index.coffee:76:4)
  at /home/nexus/app/modules/http/jsonrpc.coffee:226:5
From previous event:
  at runRequest (/home/nexus/app/modules/http/jsonrpc.coffee:225:4)
  at /home/nexus/app/modules/http/jsonrpc.coffee:295:5
  at bound (domain.js:287:14)
  at runBound (domain.js:300:12)
  at processImmediate [as _immediateCallback] (timers.js:383:17)
From previous event:
  at Server.handle (/home/nexus/app/modules/http/jsonrpc.coffee:289:4)
  at next (/home/nexus/node_modules/restify/lib/server.js:906:30)
  at f (/home/nexus/node_modules/restify/node_modules/once/once.js:17:25)
  at Server.<anonymous> (/home/nexus/app/modules/http/index.coffee:49:5)
  at next (/home/nexus/node_modules/restify/lib/server.js:906:30)
  at f (/home/nexus/node_modules/restify/node_modules/once/once.js:17:25)
  at Server.<anonymous> (/home/nexus/app/modules/http/index.coffee:35:5)
  at next (/home/nexus/node_modules/restify/lib/server.js:906:30)
  at f (/home/nexus/node_modules/restify/node_modules/once/once.js:17:25)
  at Server.parseAuthorization (/home/nexus/node_modules/restify/lib/plugins/authorization.js:146:17)
  at next (/home/nexus/node_modules/restify/lib/server.js:906:30)
  at f (/home/nexus/node_modules/restify/node_modules/once/once.js:17:25)
  at parseJson (/home/nexus/node_modules/restify/lib/plugins/json_body_parser.js:65:9)
  at Server.parseBody (/home/nexus/node_modules/restify/lib/plugins/body_parser.js:90:13)
  at next (/home/nexus/node_modules/restify/lib/server.js:906:30)
  at f (/home/nexus/node_modules/restify/node_modules/once/once.js:17:25)
  at IncomingMessage.done (/home/nexus/node_modules/restify/lib/plugins/body_reader.js:121:13)
  at IncomingMessage.g (events.js:260:16)
  at emitNone (events.js:67:13)
  at IncomingMessage.emit (events.js:166:7)
  at endReadableNT (_stream_readable.js:913:12)
  at nextTickCallbackWith2Args (node.js:442:9)
  at process._tickDomainCallback (node.js:397:17)
```

**Comment by elhigu** [comment]
I think that fix looks fine, but it would need test case too.

**Comment by funnylookinhat** [comment]
@elhigu I'm having trouble getting the actual warning to throw, but I think this causes the same issue... it prints `Unhandled rejection undefined` to the console.

```
Bluebird.resolve()
.then ->
  new Bluebird (resolve, reject) ->
    knex.transaction (trx) ->
      Bluebird.resolve()
      .then ->
        console.log "foo"
      .then ->
        throw new Error "Some error"
      .then ->
        trx.commit()
      .then ->
        resolve "bar"
      .catch (internalErr) ->
        trx.rollback()
        .catch (rollbackErr) ->
          console.log "Rollback Error: #{rollbackErr}"
        .then ->
          reject internalErr
.then (result) ->
  console.log "Done: #{result}"
.catch (err) ->
  console.log "Error: #{err}"
.then ->
  process.exit 0
```

**Comment by funnylookinhat** [comment]
Resolved in #1538
